### PR TITLE
Support url query parameters

### PR DESCRIPTION
**Pull request by otamachan**
Hi,

I'm using Atlassian Stash to manage my own ``rosdsitro``.
Unfortunately, in order to get a raw content of a file from Stash, I need to add query parameters like ``raw`` and ``at=``.
(http://stackoverflow.com/questions/14063008/get-raw-file-content-using-stash-rest-api)
Although there is already an issue, nothing has seemed to progress so far(https://jira.atlassian.com/browse/BSERV-4036) .

I would appreciate if url query parameters are supported, that is if ``index_url`` has query parameters then all the relative urls have the same queries.

Thanks.

### CODE REVIEW

**Comment by wjwwood** [comment]
Code review looks ok to me, and by the way, thanks for including a test.

I'd need to see this tested with all of our tools to make sure nothing else is negatively affected, before I'd give it a +1. Not sure when we'll get around to that, as several people will be on vacation for the next few weeks.

**Comment by dirk-thomas** [comment]
This changes the semantic of the public API. How should we move forward on this?

**Comment by otamachan** [comment]
Thank you for your feedback.
Keeping the ``base_url`` parameter and adding an extra keyword parameter like ``query`` is acceptable?

**Comment by dirk-thomas** [comment]
@otamachan Can you please give an estimate when you will update the PR? If it will be pretty soon (today or early tomorrow) we can release it with the next patch version. Otherwise it would release a patch version without it and it will go into the next one (which is uncertain when that will be).

**Comment by otamachan** [comment]
＠dirk-thomas I'll update the PR from now. It will ends within an hour.

**Comment by dirk-thomas** [comment]
Great. Thanks.

**Comment by otamachan** [comment]
I updated the PR. Please review it.

File: src/rosdistro/__init__.py [hunk 1]
```diff
@@ -49,6 +49,10 @@
     from io import BytesIO as StringIO
 import sys
 import yaml
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
```
Comment (dirk-thomas) [inline src/rosdistro/__init__.py]: Please move this block on line higher (above `yaml`) to keep the import order alphabetically.

File: test/test_index.py [hunk 2]
```diff
@@ -41,3 +41,34 @@ def test_get_index_v3_invalid():
         assert False
     except AssertionError:
         pass
+
+
+def test_get_index_from_http_with_query_parameters():
+    import time
```
Comment (dirk-thomas) [inline test/test_index.py]: Please move this line two lines lower (below `sys`) to keep the import order alphabetically.

File: test/test_index.py [hunk 3]
```diff
@@ -41,3 +41,34 @@ def test_get_index_v3_invalid():
...
+    import subprocess
+    import sys
+    url = 'http://localhost:9876/index_v3.yaml?raw&at=master'
+    # start a http server and wait
+    if sys.version_info < (3, 0, 0):
+        proc = subprocess.Popen(['python', '-m', 'SimpleHTTPServer', '9876'],
```
Comment (dirk-thomas) [inline test/test_index.py]: Instead of using `python` please use `sys.executable`.

Same below.

**Comment by dirk-thomas** [comment]
Beside the small comments the patch looks good to me.

**Comment by otamachan** [comment]
I'll fix soon.

**Comment by otamachan** [comment]
Done. Please review it.

**Comment by dirk-thomas** [comment]
Thanks for  iterating on this. +1 @wjwwood ?

**Comment by wjwwood** [comment]
lgtm

**Pull request merged**